specutils/spectra/spectrum_mixin.py

```diff
@@ -58,7 +58,13 @@ def spectral_axis(self):
         """
         Returns a Quantity array with the values of the spectral axis.
         """
-        return self.wcs.pixel_to_world(np.arange(self.flux.shape[-1]))
+
+        if len(self.flux) > 0:
+            spectral_axis = self.wcs.pixel_to_world(np.arange(self.flux.shape[-1]))
+        else:
+            spectral_axis = []*self.wcs.spectral_axis_unit
```

Comment:
In the future APE14 WCS world, I'm not sure this is going to be the same as what `pixel_to_world` should yield.  I wonder if it's safer to do something like this:
```
else:
    spectral_axis = self.wcs.pixel_to_world([0])[1:]
```

it's a bit of a hack but it seems like it might be more consistent with what `pixel_to_world` is doing otherwise?

Reply:
tbh, I would suggest we live this one as is and we can come back to change/check this another time.

Reply:
Thinking about this again, the current implementation definitely will *not* work for some WCS's in the APE14 world, as there is not guarantee the output of `pixel_to_world` is a regular quantity object (although it *is* guaranteed to follow the indexing convention).  So I'm sticking with this being the safer way to do it.  My reasoning is basically that this is the sort of thing no one checks because it's a corner case. My reasoning is basically that this is the sort of thing no one checks because it's a corner case that will only show up for relatively unusual WCS's (which we aren't exhaustively testing for)

Although now that I've written that, I realize the alternate (possibly better?) solution is to specifically write the test to ensure the `spectral_axis` of the "empty spectrum" is of the same type as the output of the non-empty one, over all the tested input data sets.  That might be too much of a pain though?

Reply:
@eteq I am seeing exceptions, did you try it?

```
spectrum = Spectrum1D(spectral_axis=[]*u.um, flux=[]*u.Jy)
spectrum.wcs.pixel_to_world([0])[1:]
```
that fails with exceptions on bounding box etc etc. 

The first paragraph (and previous comment suggests changing the empty spectral_axis output. But the last paragraph suggests just changing the tests. Sorry, I am confused.  Should the code change or the tests. (Might be easier to discuss in person).